Detectors/TPC/qc/src/PID.cxx

```diff
@@ -137,11 +193,10 @@ bool PID::processTrack(const o2::tpc::TrackTPC& track)
 void PID::dumpToFile(const std::string filename)
 {
   auto f = std::unique_ptr<TFile>(TFile::Open(filename.c_str(), "recreate"));
-  for (auto& hist : mHist1D) {
-    f->WriteObject(&hist, hist.GetName());
-  }
-  for (auto& hist : mHist2D) {
-    f->WriteObject(&hist, hist.GetName());
+  for (const auto& pair : mapHist2D) {
+    for (auto& hist : pair.second) {
+      f->WriteObject(&hist, hist->GetName());
+    }
```

Comment:
It might be nice to keep the grouping in the vectors, instead of writing all histograms in a flat way
```suggestion
  for (const auto& [name, histos] : mapHist2D) {
    TObjArray arr;
    arr.SetName(name.data());
    for (auto& hist : histos) {
      arr.Add(hist.get());
    }
    arr.Write(arr.GetName(), TObject::kSingleKey);
```